### PR TITLE
imp(invoices): Support line break on invoice footer

### PR DESCRIPTION
**Pull request by jdenquin**
## Description

- Add the line break support into the invoice PDF for invoice footer

### CODE REVIEW

File: app/services/utils/pdf_generator.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require_relative '../../views/helpers/line_break_helper.rb'
```
Comment (jdenquin) [inline app/services/utils/pdf_generator.rb:3]: Style/RedundantFileExtensionInRequire: Redundant `.rb` file extension detected. (https://rubystyle.guide#no-explicit-rb-to-require)

File: app/views/helpers/line_break_helper.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class LineBreakHelper
+  def self.break_lines(text)
+    text.to_s.gsub('\n', '<br/>').html_safe
```
Comment (jdenquin) [inline app/views/helpers/line_break_helper.rb:5]: Rails/OutputSafety: Tagging a string as html safe may be a security risk.

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -28,6 +28,7 @@
 ActiveJob::Uniqueness.test_mode!
 
 Dir[Rails.root.join('spec', 'support', '**', '*.rb')].sort.each { |f| require f }
+Dir[Rails.root.join('app', 'views', 'helpers', '*.rb')].sort.each { |f| require f }
```
Comment (jdenquin) [inline spec/rails_helper.rb:31]: Rails/FilePath: Prefer `Rails.root.join('path/to')`.

File: app/services/utils/pdf_generator.rb (hunk 1 above)
Comment (vincent-pochet) [reply app/services/utils/pdf_generator.rb]: I suggest to add the `helper_folder` in `config.eager_load_paths` (in `/config/application.rb`)

File: app/views/helpers/line_break_helper.rb (hunk 2 above)
Comment (vincent-pochet) [reply app/views/helpers/line_break_helper.rb]: I think if we `html_safe` it, we also have to sanitize it before, just to be safe

File: app/services/utils/pdf_generator.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require_relative '../../views/helpers/line_break_helper'
```
Comment (vincent-pochet) [inline app/services/utils/pdf_generator.rb:3]: I suggest to add the helper_folder in config.eager_load_paths (in /config/application.rb)

File: app/views/helpers/line_break_helper.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class LineBreakHelper
+  def self.break_lines(text)
+    escaped_text = ERB::Util.html_escape(text)
+    escaped_text.to_s.gsub(/\n/, '<br/>').html_safe
+  end
+end
```
Comment (jdenquin) [inline app/views/helpers/line_break_helper.rb:8]: Rails/OutputSafety: Tagging a string as html safe may be a security risk.

File: app/views/helpers/line_break_helper.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class LineBreakHelper
+  def self.break_lines(text)
+    escaped_text = ERB::Util.html_escape(text)
+    escaped_text.to_s.gsub(/\n/, '<br/>').html_safe
```
Comment (jdenquin) [inline app/views/helpers/line_break_helper.rb:6]: Rails/OutputSafety: Tagging a string as html safe may be a security risk.

File: app/views/helpers/line_break_helper.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class LineBreakHelper
+  def self.break_lines(text)
+    escaped_text = ERB::Util.html_escape(text)
+    escaped_text.to_s.gsub(/\n/, '<br/>').html_safe # rubocop:disable Rails/OutputSafety
```
Comment (vincent-pochet) [inline app/views/helpers/line_break_helper.rb:6]: I guess the `to_s` should be in the previous line right?

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**